### ISSUE TITLE
Unable to close the account

### ISSUE THREAD

**Issue by Pantamis** [lightninglabs/pool#163]
Hi !

I have issue to close a pool account with LiT 0.3.0. I had no connection issue while doing what I did.

What I have done exactly:

- I canceled all my (ask) orders with 
`pool --rpcserver=localhost:10009 orders cancel [nonce]`
- I checked that my order was indeed canceled (`"state": "ORDER_CANCELED"`) with
`pool --rpcserver=localhost:10009 orders list`
- My account was in the state `"OPEN"`, I checked it with `accounts list`
- Then I closed the account with
`pool --rpcserver=localhost:10009 accounts close 02[hidden key]4 --sat_per_vbyte 1`
It return an error: `[pool] rpc error: code = Unknown desc = no order found`
In logs I found the following lines: 
`RPCS: Closing acct=02[hidden key]4`
`[/poolrpc.ChannelAuctioneer/OrderState]: rpc error: code = Unknown desc = no order found`

The status of my account: not closed, the last txid is the same than before (`accounts list` tell me that), same amount of money inside the account. 

However, I can still submit ask orders with my account. I will update this issue if I successfully match an opposite order and open channels with my account.

It seems there is an issue with closing my account anyway, is there an explanation of the error returned by pool client ?

**Comment by Roasbeef** [comment]
Can you try `pool orders list --show_archived`, are any of the order seen there amongst the one that isn't found from the client's PoV?

**Comment by Roasbeef** [comment]
This patch locally _should_ let you close the account: 
```diff
diff --git a/rpcserver.go b/rpcserver.go
index 7012082..dfb8611 100644
--- a/rpcserver.go
+++ b/rpcserver.go
@@ -924,7 +924,8 @@ func (s *rpcServer) CloseAccount(ctx context.Context,
 		// the auctioneer's state.
 		orderStateResp, err := s.auctioneer.OrderState(ctx, nonce)
 		if err != nil {
-			return nil, err
+			log.Errorf("unable to get state for order=%v: %v", nonce, err)
+			continue
 		}
 		orderState, err := rpcOrderStateToDBState(orderStateResp.State)
 		if err != nil {

```

It's based on the latest client version (v0.3.3) would recommend you update to it if you aren't on it already.


However I'm still interested in that question above, so we can attempt to resolve this issue on a fundamental level if it's something else.

**Comment by Pantamis** [comment]
Is it safe to run LiT and another standalone version of pool at the same time ?

It seems that my submitted orders are all correctly there. `pool orders list` (no `--show_archived` flag in my version 0.3.0) return me the list of all orders:

- the 2 canceled orders I made are shown
- I have one submitted order which failed because amount was not a multiple of 100 000 satoshi
- I have submitted a new order which is now in state "ORDER_SUBMITTED" . I am fine with letting this order running for now, I will notify you if it is partially executed.
- I also ran the submit command several time but refused to confirm the order submssion, no order are shown for this case (I guess it is normal since they should not have nonce I guess)

If nothing happen by the end of the week, I will try to close the account again with the new pool version (if safe to run with LiT) and your patch.

**Comment by guggero** [comment]
If you configure the ports and data directories to be different, you can run multiple independent instances of pool. But if you run with the default settings, one will probably not start correctly because the other already has the unique lock on the database.
If you ask if you could switch between LiT and standalone Pool for the same accounts/orders, I would strongly advise against it, unless you use the exact same versions for both.

**Comment by Pantamis** [comment]
Little update:

- I found logs saying my order was matched yesterday by the auctioner but it had never been executed for unclear reasons

- I stopped LiT to install the new release 0.3.1 with 0.3.3 version of pool.

- I restarted LiT

 - Later, my order was executed successfully

So it seems the version of pool plays a role. I won't try to close the account for now because I will continue to use it.

I close the issue for now, I reopened it if I still have trouble to close my account next time I want to.

Thanks for the help !
